sources/lib/structures/typed.ml

```diff
@@ -234,6 +239,14 @@ let rec print_term fmt t = match t.c.tt_desc with
   | TTite(cond, t1, t2) ->
     fprintf fmt "(if %a then %a else %a)"
       print_formula cond print_term t1 print_term t2
+  | TTproject (grded, t1, s) ->
+    fprintf fmt "%a#%s%s"
+      print_term t1 (if grded then "" else "!") (Hstring.view s)
+
+  | TTmatch _ -> assert false
```

Comment:
As long as we keep the typed AST as the common point between all frontend/input methods, it might be interesting to be able to print it, if only for debugging purposes.